src/formatDate.js

```diff
@@ -50,7 +50,36 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dataObj = {};
+  const dataFromArr = date.split(fromFormat[3]);
+  const dataToArr = [];
+  const separator = toFormat[3];
+
+  const fromYYtoYYYY = (obj) => {
+    if (obj['YY'] < 30) {
+      obj['YYYY'] = '20' + obj['YY'];
+    } else {
+      obj['YYYY'] = '19' + obj['YY'];
+    }
+  };
```

Comment:
When you creating helper function - move it outside parent functions (there is no need to create this function every time when parent function is called)